src/Microsoft.Identity.Client/Core/ServiceBundle.cs

```diff
@@ -33,10 +33,10 @@ internal class ServiceBundle : IServiceBundle
             PlatformProxy = PlatformProxyFactory.CreatePlatformProxy(DefaultLogger);
             HttpManager = config.HttpManager ?? new HttpManager(config.HttpClientFactory);
 
-            if (config.MatsConfig != null)
+            if (config.TelemetryConfig != null)
             {
                 // This can return null if the device isn't sampled in.  There's no need for processing MATS events if we're not going to send them.
-                Mats = MatsTelemetryClient.CreateMats(config, PlatformProxy, config.MatsConfig);
+                Mats = MatsTelemetryClient.CreateMats(config, PlatformProxy, config.TelemetryConfig);
```

Comment:
rename?

Reply:
These are internal classes and for now we have TelemetryCore as a namespace which is all of the "raw" events that are processed by the MATS-like infrastructure.  Yes, we need to look at naming and cleanup here (which is thankfully internal) but folks today know MATS and this makes it clear that this is the MATS portion of the code while we resolve our final story and also do some future refactoring around telemetrymanager and this piece (which i'd rename telemetryclient).  do folks think having telemetrymanager (for raw events) and telemetryclient (for mats) is confusing?  Or should I just move all of the mats namespace into TelemetryCore and call this TelemetryClient and be done with it?

Reply:
i vote for the second option ( move all of mats into Telemetrycore)

---
In reply to: [287118878](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/1161#discussion_r287118878) [](ancestors = 287118878)